google/cloud/options_test.cc

```diff
@@ -328,6 +328,28 @@ TEST(OptionsSpan, Basics) {
   EXPECT_FALSE(internal::CurrentOptions().has<IntOption>());
 }
 
+TEST(FetchOption, Basics) {
+  auto opts = Options{}.set<StringOption>("foo").set<IntOption>(42);
+
+  auto b = internal::FetchOption<BoolOption>(opts);
+  EXPECT_FALSE(opts.has<BoolOption>());
+  EXPECT_TRUE(opts.has<IntOption>());
+  EXPECT_TRUE(opts.has<StringOption>());
```

Comment:
Move these three groups of three-repeated expectations to immediately after the initialization of `opts`.